src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -43,77 +44,71 @@ class VisitService(
-        supportList.forEach {
-          if (!supportTypeRepository.existsByName(it.type)) {
-            throw SupportNotFoundException("Invalid support ${it.type} not found")
-          }
-          visitEntity.support.add(createVisitSupport(visitEntity, it.type, it.text))
+    reserveVisitSlotDto.visitorSupport?.let { supportList ->
+      supportList.forEach {
+        if (!supportTypeRepository.existsByName(it.type)) {
+          throw SupportNotFoundException("Invalid support ${it.type} not found")
         }
+        visitEntity.support.add(createVisitSupport(visitEntity, it.type, it.text))
       }
-
-      createVisitTrackEvent(visitEntity)
-      return VisitDto(visitEntity)
-    } else {
-      return changeVisitSlot(bookedVisit!!.applicationReference, mapToChangeVisitSlotRequestDto(reserveVisitSlotDto))
     }
-  }
 
-  private fun mapToChangeVisitSlotRequestDto(reserveVisitSlotDto: ReserveVisitSlotDto): ChangeVisitSlotRequestDto {
-    return ChangeVisitSlotRequestDto(
-      visitRestriction = reserveVisitSlotDto.visitRestriction,
-      visitContact = reserveVisitSlotDto.visitContact,
-      visitors = reserveVisitSlotDto.visitors,
-      visitorSupport = reserveVisitSlotDto.visitorSupport
-    )
+    createVisitTrackEvent(visitEntity)
+    return VisitDto(visitEntity)
   }
 
-  private fun isVisitNew(bookedVisit: Visit?, reserveVisitSlotDto: ReserveVisitSlotDto): Boolean {
+  private fun getStartingStatus(bookingReference: String, prisonId: String? = null, prisonerId: String? = null, startTimestamp: LocalDateTime): VisitStatus {
+
+    val bookedVisit = this.visitRepository.findBookedVisit(bookingReference)
 
     if (bookedVisit == null ||
-      bookedVisit.prisonId != reserveVisitSlotDto.prisonId ||
-      bookedVisit.prisonerId != reserveVisitSlotDto.prisonerId ||
-      bookedVisit.visitStart.compareTo(reserveVisitSlotDto.startTimestamp) != 0
+      (prisonId != null && bookedVisit.prisonId != prisonId) ||
+      (prisonerId != null && bookedVisit.prisonerId != prisonerId) ||
+      bookedVisit.visitStart.compareTo(startTimestamp) != 0
     ) {
-      return true
+      return RESERVED
     }
-
-    return false
+    return CHANGING
   }
 
   fun changeVisitSlot(applicationReference: String, changeVisitSlotRequestDto: ChangeVisitSlotRequestDto): VisitDto {
     val visitEntity = visitRepository.findByApplicationReference(applicationReference) ?: throw VisitNotFoundException("Reserved visit reference $applicationReference not found")
 
     changeVisitSlotRequestDto.visitRestriction?.let { visitRestriction -> visitEntity.visitRestriction = visitRestriction }
-    changeVisitSlotRequestDto.startTimestamp?.let { visitStart -> visitEntity.visitStart = visitStart }
+    changeVisitSlotRequestDto.startTimestamp?.let {
+      visitEntity.visitStart = it
+      visitEntity.visitStatus = getStartingStatus(visitEntity.reference, startTimestamp = changeVisitSlotRequestDto.startTimestamp)
```

Comment:
just a query, do we need this outside the loop?

Reply:
I did think about this, the only thing that can change status at this stage is the startTimestamp if it's null it has to change since the step so no status needs to be updates,.. if it has change the status might need to change.. I think we are good